### DISCUSSION TITLE
'--dry-run -oyaml' feature in argocd CLI to print Application, ApplicationSet and AppProject

### DISCUSSION THREAD

**Discussion by jaythamke** [argoproj/argo-cd#16257] [Q&A]
Hi,
Executing `argocd app create ...` directly creates the Application and deploys into a k8s cluster. 
I did not find the option to create a manifest of Application CR before creating the CR in the cluster.
How to create such manifest (YAML) files.
Also asked the query on Slack channel (Kubernetes>gitops) - https://kubernetes.slack.com/archives/CBT6N1ASG/p1699306997207369
Thanks,
Jayesh

**Comment by iheanacho-chukwu** [comment]
Update: This seems to relate closely to enhancement request #23280: https://github.com/argoproj/argo-cd/issues/23280

Feel free to upvote or comment there to support the feature.

**Comment by Angelin01** [comment]
Similarly, being able to run `argocd app diff`, even for applications that don't exist, would be very convenient.

One might immediately think: "Why do you need to diff an App that doesn't exist yet?"

It's for CI/CD pipelines that deploy to multiple environments! Commonly, you deploy to development environments first and only deploy to production later. Being able to see the diff at first deployment time, even if it shows everything as being "created", is a good sanity check and helps ensure you are not missing something.
